base/compiler/tfuncs.jl

```diff
@@ -727,17 +727,21 @@ function getfield_tfunc(@nospecialize(s00), @nospecialize(name))
             end
         end
         s = typeof(sv)
-    elseif isa(s, PartialStruct)
+    elseif isa(s00, PartialStruct)
+        s01 = widenconst(s00)
+        s = unwrap_unionall(s01)::DataType
```

Comment:
```suggestion
        s = s01::DataType
```
currently this simplification would hold, but do we want to assume that?